holidays/countries/china.py

```diff
@@ -9,53 +9,375 @@
+
+        # 国庆节
+        # Status: In-Use (Statutory).
+        # Oct 1-2 in 1949, 1999, 2007, and 2013 revision
+        # Additional Holiday (Oct 3) is available from Sep 1999 (1999 rev.).
+
+        # National Day Golden Weekend
+        # Checked with Official Notice from 2001-2023.
+
+        # National Day.
+        national_day = tr("国庆节")
+        self._add_holiday_oct_1(national_day)
+        self._add_holiday_oct_2(national_day)
+        if self._year >= 1999:
+            self._add_holiday_oct_3(national_day)
+        if self.observed and 2000 <= self._year != 2008:
+            # Non-Statutory.
+            self._add_holiday_oct_4(national_day)
+            self._add_holiday_oct_5(national_day)
+            self._add_holiday_oct_6(national_day)
+            if self._year != 2023:
+                self._add_holiday_oct_7(national_day)
+
+        if self._year >= 2008:
+            # 清明节
+            # Status: In-Use (Statutory).
+            # Tomb-Sweeping Day in 2007, and 2013 revision.
+            # Consecutive Holidays are available from 2008, except in 2014/2015/2016/2019.
+
+            # Tomb-Sweeping Day.
+            qingming_festival = self._add_qingming_festival(tr("清明节"))
+
+            if self.observed and self._year not in {2014, 2015, 2016, 2019, 2023}:
+                # Non-Statutory.
+                self._add_observed_one_off_holidays(qingming_festival)
+
+            # 端午节
+            # Status: In-Use (Statutory).
+            # Dragon Boat Festival in 2007, and 2013 revision.
+            # Consecutive Holidays are available from 2008, except in 2014/2015/2018/2019/2023.
+
+            # Dragon Boat Festival.
+            dragon_boat_festival = self._add_dragon_boat_festival(tr("端午节"))
+
+            if self.observed and self._year not in {2014, 2015, 2018, 2019}:
+                # Non-Statutory.
+                self._add_observed_one_off_holidays(dragon_boat_festival)
+
+            # 中秋节
+            # Status: In-Use (Statutory).
+            # Mid-Autumn Festival in 2007, and 2013 revision.
+            # Consecutive Holidays are available from 2008, except in 2010/2014/2015/2018/2019.
+            # Extra Day (Oct 8) is instead aded to the National Day Week if overlaps.
+
+            # Mid-Autumn Festival.
+            mid_autumn_festival = self._add_mid_autumn_festival(tr("中秋节"))
+
+            if self.observed:
+                # Non-Statutory.
+                if (
+                    mid_autumn_festival == date(self._year, OCT, 1)
+                    or mid_autumn_festival == date(self._year, OCT, 2)
+                    or mid_autumn_festival == date(self._year, OCT, 3)
+                    or mid_autumn_festival == date(self._year, OCT, 4)
+                    or mid_autumn_festival == date(self._year, OCT, 5)
+                    or mid_autumn_festival == date(self._year, OCT, 6)
+                    or mid_autumn_festival == date(self._year, OCT, 7)
+                ):
```

Comment:
No idea why this comment is missing from the previous round:
```suggestion
                if mid_autumn_festival.month == OCT and 1 <= mid_autumn_festival.day <= 7:
```